dashboard/app/controllers/api_controller.rb

```diff
@@ -442,7 +442,27 @@ def course_summary
 
   def unit_summary
     unit_name = params[:unit_name]
-    unit = Unit.get_from_cache(unit_name)
+    unit_position = params[:unit_position]
+    course_name = params[:course_name]
+    unit = nil
+
+    # unit_summary supports calls with either :unit_name or :course_name AND :unit_position
```

Comment:
Could we add some validation that if `unit_name` is supplied, then `unit_position` and `course_name` are nil and vice versa? I think that would be better than a comment

Reply:
I have added parameter validation and unit tests.